### PR TITLE
Add some r/w functionality to GCP Bigtable Hook

### PR DESCRIPTION
**Pull request by EmilyMazo**
Make sure you have checked _all_ steps below.

### Jira

- [ X] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-6155

### Description

- [ X] Here are some details about my PR, including screenshots of any UI changes:
The GCP Bigtable Hook, a wrapper about the GCP Client functionality for Bigtable, has only a few methods currently, all addressing administrative tasks. I would like to add the ability to read and write Bigtable data in rows using the Hook.

### Tests

- [X ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
tests/gcp/hooks/test_bigtable/test_create_rowkey_regex_filter
tests/gcp/hooks/test_bigtable/test_delete_row
tests/gcp/hooks/test_bigtable/test_check_and_mutate_row

### Commits

- [ X] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ X] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### CODE REVIEW

File: airflow/gcp/hooks/bigtable.py [hunk 1]
```diff
@@ -276,3 +278,58 @@ def get_cluster_states_for_table(instance: Instance, table_id: str) -> Dict[str,
 
         table = Table(table_id, instance)
         return table.get_cluster_states()
+
+    def create_rowkey_regex_filter(self, regex):
```
Comment (turbaszek) [inline airflow/gcp/hooks/bigtable.py]: Where is the helper used?

File: airflow/gcp/hooks/bigtable.py [hunk 2]
```diff
@@ -276,3 +278,58 @@ def get_cluster_states_for_table(instance: Instance, table_id: str) -> Dict[str,
...
+        will only match rows with keys that exactly match the regex string.
+        Returns a RowFilter.
+
+        :type regex: str
+        :param regex: The exact match regex string for the value we wish to find
+        """
+        return RowKeyRegexFilter(regex)
+
+
+    def delete_row(self, instance, table_id, row_key):
```
Comment (turbaszek) [inline airflow/gcp/hooks/bigtable.py]: Please add type annotations.
```suggestion
    def delete_row(self, instance: Instance, table_id: str, row_key: str) -> None:
```

**Review by turbaszek — CHANGES_REQUESTED** [review]

File: airflow/gcp/hooks/bigtable.py (hunk 1 above)
Comment (EmilyMazo) [reply airflow/gcp/hooks/bigtable.py]: This helper (and the new one like it) is only used outside the Hook, for creating RowFilter objects that can be passed in to the `check_and_mutate_row` method.

**Comment by OmerJog** [comment]
@EmilyMazo can you please fix errors

```
airflow/gcp/hooks/bigtable.py:260:111: E501 line too long (138 > 110 characters)
airflow/gcp/hooks/bigtable.py:270:111: E501 line too long (120 > 110 characters)
airflow/gcp/hooks/bigtable.py:277:111: E501 line too long (137 > 110 characters)
airflow/gcp/hooks/bigtable.py:288:111: E501 line too long (120 > 110 characters)
airflow/gcp/hooks/bigtable.py:315:16: F821 undefined name 'ValueRegexFilter'
airflow/gcp/hooks/bigtable.py:318:5: E303 too many blank lines (2)
airflow/gcp/hooks/bigtable.py:318:111: E501 line too long (112 > 110 characters)
airflow/gcp/hooks/bigtable.py:328:111: E501 line too long (120 > 110 characters)
airflow/gcp/hooks/bigtable.py:337:111: E501 line too long (213 > 110 characters)
airflow/gcp/hooks/bigtable.py:337:129: F821 undefined name 'RowFilter'
airflow/gcp/hooks/bigtable.py:339:111: E501 line too long (111 > 110 characters)
airflow/gcp/hooks/bigtable.py:356:111: E501 line too long (113 > 110 characters)
airflow/gcp/hooks/bigtable.py:360:111: E501 line too long (120 > 110 characters)
tests/gcp/hooks/test_bigtable.py:42:1: E302 expected 2 blank lines, found 1
tests/gcp/hooks/test_bigtable.py:482:111: E501 line too long (148 > 110 characters)
tests/gcp/hooks/test_bigtable.py:482:141: E999 SyntaxError: positional argument follows keyword argument
```

Also note that the Jira in your commit message is wrong.
It should be `AIRFLOW-6155` not `AIRFLOW-6713`

**Comment by EmilyMazo** [comment]
Very sorry about the repeat commits to fix type annotations and other issues that would be picked up by testing locally- I'm having a bear of a time setting up the local testing environment. My apologies for the clutter! I can rebase and squash all of these commits when I get this cleaned up.

**Pull request closed without merging**